frontend/src/lib/components/metadata/cells/metadata-cells/ContinuousMetadataCell.svelte

```diff
@@ -17,57 +17,47 @@
 	export let updatePredicates: (predicates: FilterPredicate[]) => void;
 
 	let view: View;
-	let localSelection: number[] = [];
-	let localSelectionX: number[] = [];
 
-	function updateSel(filterPreds: FilterPredicate[]) {
-		if (filterPreds.length !== 0) {
-			view.signal('brush', { bucket: localSelection });
-			view.signal('brush_x', localSelectionX);
-		} else {
-			view.signal('brush', {});
+	$: if (view) {
+		view.removeSignalListener('brush', setSelection);
+		if (filterPredicates.length === 0) {
 			view.signal('brush_x', []);
 		}
-		view.runAsync();
-	}
-
-	$: if (view) {
-		view.addSignalListener('brush', (...s) => (localSelection = s[1].bucket ? s[1].bucket : []));
-		view.addSignalListener('brush_x', (...s) => (localSelectionX = s[1]));
+		view.addSignalListener('brush', setSelection);
 	}
 
-	$: if (view) {
-		updateSel(filterPredicates);
-	}
-
-	function setSelection() {
-		if (localSelection.length > 0) {
+	let timer: ReturnType<typeof setTimeout>;
```

Comment:
Nit: maybe move to the other lets? Don't know how we do it generally? I normally do exports then vars then reactive then funcs but open to do this differently.